app/models/category.rb

```diff
@@ -62,4 +65,19 @@ def self.get_cached_categories_grouped_by_parent
   def self.ransackable_attributes(auth_object = nil)
     ["description", "name", "related_terms"]
   end
+
+  def self.prepared_categories_for_practice_editor(is_admin)
+    get_parent_categories(is_admin).each_with_object({}) do |parent_category, hash|
+      categories = parent_category.sub_categories.where(is_other: false).order_by_name.to_a
+
+      if categories.any? && parent_category.name != "Communities"
+        all_cat = new(name: "All #{parent_category.name.downcase}", parent_category: parent_category)
+        other_cat = new(name: 'Other', parent_category: parent_category)
+        categories.prepend(all_cat)
+        categories.append(other_cat)
+      end
```

Comment:
this conditional in the middle of the method is to ensure that an "All" and "Other" button do not appear in the options under the "Communities" category. 

The choice to exclude the "Other" button I think was a safe bet as it will prevent confusing additions to the "Communities" `parent_category`, but excluding the "All" button was a liberty I took because including it would mean adjusting the JS and it didn't seem to me to be particularly useful given that there will be so few `categories` under "Communities"

Reply:
Good choice. I agree with your logic here, especially if we head towards eliminating `Other` categories.